File: src/services/email.ts
import nodemailer from 'nodemailer'

export async function sendEmailHtml(to: string, subject: string, html: string) {
  const host = process.env.SMTP_HOST || ''
  const port = Number(process.env.SMTP_PORT || 465)
  const user = process.env.SMTP_USER || ''
  const pass = process.env.SMTP_PASS || ''
  if (!host || !user || !pass) {
    return { ok: false, error: 'smtp_not_configured' }
  }
  const transporter = nodemailer.createTransport({
    host,
    port,
    secure: port === 465,
    auth: { user, pass }
  })
  try {
    const info = await transporter.sendMail({ from: user, to, subject, html })
    return { ok: true, id: info.messageId }
  } catch (e: any) {
    return { ok: false, error: String(e?.message || e) }
  }
}

File: app/services/email.py
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
from app.config import settings
from app.utils.logger import logger


async def send_email(to_address: str, content: str, title: str = "每周Newsletter") -> dict:
    """Send email via SMTP"""
    if not all([settings.smtp_host, settings.smtp_user, settings.smtp_pass]):
        logger.warning("email_not_configured")
        return {"ok": False, "error": "Email not configured"}

    try:
        msg = MIMEMultipart("alternative")
        msg["Subject"] = title
        msg["From"] = settings.smtp_user
        msg["To"] = to_address

        # Attach HTML content
        html_part = MIMEText(content, "html", "utf-8")
        msg.attach(html_part)

        # Send email
        with smtplib.SMTP_SSL(settings.smtp_host, settings.smtp_port) as server:
            server.login(settings.smtp_user, settings.smtp_pass)
            server.send_message(msg)

        logger.info("email_sent", to=to_address)
        return {"ok": True}

    except Exception as e:
        logger.error("email_failed", to=to_address, error=str(e))
        return {"ok": False, "error": str(e)}
